### PR TITLE
[FLINK-13210][hive] Hive connector test should dependent on blink planner instead of legacy planner

### PR DESCRIPTION
**Pull request by lirui-apache**
## What is the purpose of the change

To test hive connector with blink planner instead of legacy planner.

## Brief change log

  - Depend on blink planner instead.
  - Updated `HiveTableSourceTest` and `HiveTableSinkTest` to use blink planner.
  - Changed `HiveTableFactoryTest` to stop testing with CSV table. Because CSV table factories are in flink planner and we can't depend on both of the planners right now.

## Verifying this change

Existing and modified test cases.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? NA

### CODE REVIEW

**Comment by lirui-apache** [comment]
@JingsongLi @godfreyhe @xuefuz @bowenli86 @zjuwangg please take a look. Thanks.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveTableSinkTest.java [hunk 1]
```diff
@@ -83,14 +92,14 @@ public void testInsertIntoNonPartitionTable() throws Exception {
...
+		TableEnvironment tableEnv = HiveTestUtils.createTableEnv();
 		List<Row> toWrite = generateRecords(5);
-		tableEnv.registerDataSet("src", execEnv.fromCollection(toWrite, rowTypeInfo));
+		Table src = tableEnv.fromTableSource(new CollectionTableSource(toWrite, rowTypeInfo));
+		tableEnv.registerTable("src", src);
 
 		tableEnv.registerCatalog("hive", hiveCatalog);
 		tableEnv.sqlQuery("select * from src").insertInto("hive", "default", "dest");
-		execEnv.execute();
+		tableEnv.execute(null);
```
Comment (bowenli86) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveTableSinkTest.java]: give it a test name, like "mytest"?

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveTableSinkTest.java [hunk 2]
```diff
@@ -104,14 +113,15 @@ public void testInsertIntoDynamicPartition() throws Exception {
...
+
 		List<Row> toWrite = generateRecords(5);
-		tableEnv.registerDataSet("src", execEnv.fromCollection(toWrite, rowTypeInfo));
+		Table src = tableEnv.fromTableSource(new CollectionTableSource(toWrite, rowTypeInfo));
+		tableEnv.registerTable("src", src);
 
 		tableEnv.registerCatalog("hive", hiveCatalog);
 		tableEnv.sqlQuery("select * from src").insertInto("hive", "default", "dest");
-		execEnv.execute();
+		tableEnv.execute(null);
```
Comment (bowenli86) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveTableSinkTest.java]: ditto

**Comment by lirui-apache** [comment]
@bowenli86 Thanks for the review. PR updated accordingly.

**Comment by JingsongLi** [comment]
@lirui-apache Thanks for your contribution, LGTM

**Comment by lirui-apache** [comment]
@JingsongLi thanks for the review. One side note is blink planner requires a table source to implement `getReturnType()` method (hence the change to `HiveTableSource`). This method is marked as deprecated and seems the legacy planner doesn't require it. So I think it's better if we can avoid this inconsistency.

**Comment by JingsongLi** [comment]
> getReturnType

Hi @lirui-apache , you should implement `getDataStream` and get rid of `getReturnType `

**Comment by JingsongLi** [comment]
> @JingsongLi thanks for the review. One side note is blink planner requires a table source to implement `getReturnType()` method (hence the change to `HiveTableSource`). This method is marked as deprecated and seems the legacy planner doesn't require it. So I think it's better if we can avoid this inconsistency.

Hi @lirui-apache , you are right, Blink-planner should make some changes to avoid calling `getReturnType`.
I create a JIRA: https://issues.apache.org/jira/browse/FLINK-13391 about it. Will soon fix it.

**Comment by wuchong** [comment]
As discussed offline, we will add back flink planner tests for Hive connector once FLINK-13267 is resolved. We will merge this pull request first to unblock other issues (incl. Hive UDFs).

**Comment by lirui-apache** [comment]
@wuchong Thanks for the review. PR rebased.

**Comment by lirui-apache** [comment]
@wuchong `HiveCatalogITCase ` failed in my own Travis test because of some class name conflicts. @godfreyhe is working on the issue. So let's hold on this one until that fix is in.

**Comment by wuchong** [comment]
OK @lirui-apache .

**Comment by wuchong** [comment]
Hi @lirui-apache , which issue do you depend on ?

**Comment by xuefuz** [comment]
@wuchong probably https://github.com/apache/flink/pull/9216.

**Comment by lirui-apache** [comment]
@wuchong it's #9216. I just rebased the PR

**Comment by lirui-apache** [comment]
Travis build on my own repo finished: https://travis-ci.org/lirui-apache/flink/builds/563356262
Hive connector tests passed and I don't think other failures are related to this PR

**Comment by wuchong** [comment]
Travis passed: https://travis-ci.com/flink-ci/flink/builds/120541820
Will merge this.

**Pull request closed without merging**